### ISSUE TITLE
UI Suggestions

### ISSUE THREAD

**Issue by mras0** [vAmigaWeb/vAmigaWeb#100]
Got an iPhone some time ago, and got around to checking out vAmigaWeb on it (as a PWA). Overall seems to work nicely, so great job there. A few suggestions/notes:
  - It currently defaults to ECS_2MB + 2MB chip + 2MB fast. Probably better to have it start with a more classic configuration (OCS + 512K Chip + 512K slow == WinUAE "most common"). I think this will match what most users that don't twiddle with the settings expect.
  - Might be an idea to have some kind of one-click way to select a few useful defaults (A500 expanded, A500 plain, A600, Monster (28MHz + 8MB fast)) - Don't go overboard, just a few well chosen options. Ideally this should also choose the correct KS ROM if installed (maybe warn if it isn't).
  - Add (more clear) sections to the settings dialog and move the non-advanced ones to the top (e.g. I think compatability settings should be basically hidden)
  - In firefox (on my windows PC) I get a lot of popups saying "error while open db: InvalidStateError: A mutation operation was attempted on a database that did not allow mutations." if I run vAmigaWeb in private mode (to test it without my usual settings).
  - Light/dark mode is OK, but neither seems to match my preference. In light mode the blanked area is completely white, while the area outside the app seems to be more cream colored. In dark mode the blanked area is gray while I'd prefer it to be completely black. This is just my preference, so maybe it'd be an idea to allow choosing the color if it can't be read from the system. (See screenshot below).

![Screenshot 2022-06-03 at 18 30 30](https://user-images.githubusercontent.com/9952017/171911399-723dd546-0f2f-449a-ab8a-e9cee91ced09.png)

Finally, I managed to insert a disk and watch part one of desert dream, but now I can't seem to figure out how to insert another disk. Maybe I'm just bad at iPhone, but when I click on the file slot it gives the popup menu with no way to insert another disk, and if I press on the popup it brings me to the camera or photos app seemingly at random?

Again, great job, and just some feedback. Feel free to create new sub-issues and close this one so you don't have a single, major issue open :)

**Comment by mithrendal** [comment]
great suggestions ... lets consider them one by one with some more time ...

>I can't seem to figure out how to insert another disk.
but just a quick reply on disk insertion ...

1. the thing with the file slot is that it you have to find your files ... you managed well to insert disk1 of desert dream this leads me to the think that it has maybe to do with the file extension... i.e. disk1 is .adf which is selectable by the iOS file dialog and maybe disk2 is .dms ?? and this is maybe somehow not selectable ? can this be the case ?

2. if you pack both disks into a `zip` and load that zip into vAmigaWeb then file slot becomes a zip file reader i.e. it loads the zip file at once and when you tap on the file slot (which is transformed into a zip archive) again then you can select from the zip file

3. `drag and drop`: open vAmigaWeb and switch to the iOS files app (i.e. leaving vAmigaWeb app).  In the iOS files app tap and hold the disk you want (drag) then leave the files app and switch into vAmigaWeb again and drop the disk into screen space ... this will load the disk because vAmigaWeb is a drop target

**Comment by mras0** [comment]
Did a web search on how to do drag on drop on an iPhone (yes I'm old/new), and it didn't work at first and I ended up pressing the reset button by accident. Now everything works as expected, so I think the issue I saw with it randomly choosing between camera/photos app is the app getting into a bad state (those are the first two options with the third being choose a file, maybe I could have gotten there somehow, it was not visible however).

I'll try to see if I can reproduce it, in the meantime [hardwired](https://files.scene.org/view/parties/1991/theparty91/amiga/demo/crionics-silents_hardwired.zip) from a zip file seems to work great 👍 (though it's not super intuitive for me how i'd press the right mouse button).

It also seems like when I do something (not determined yet) with the GUI there are sound issues (like some "gui" sounds is starting multiple times but getting cut off), I'll check further later on.

**Comment by Vweber73** [comment]
Hi,
Good suggestions.

One thing about the .zip files containing multiple disks would be an option to automatically use up to 4 drives, DF0: to DF3:, and insert the disks in them, in order not to act as a DJ... :)

Cheers

**Comment by Vweber73** [comment]
Excellent ideas ! :)

**Comment by mithrendal** [comment]
>In firefox (on my windows PC) I get a lot of popups saying "error while open db: InvalidStateError: A mutation operation was attempted on a database that did not allow mutations." if I run vAmigaWeb in private mode (to test it without my usual settings).

@mras0 I fixed this, it should run now also in a private / incognito browser window. Of course it does not allow to persist anything in this mode...

**Comment by mras0** [comment]
> > In firefox (on my windows PC) I get a lot of popups saying "error while open db: InvalidStateError: A mutation operation was attempted on a database that did not allow mutations." if I run vAmigaWeb in private mode (to test it without my usual settings).
> 
> @mras0 I fixed this, it should run now also in a private / incognito browser window. Of course it does not allow to persist anything in this mode...

Yes fixed, thanks. Noticed another small thing: Unmapped keys (e.g. F11) are interpreted as raw keycode 0 ("accent grave") (maybe some missing checks for whether `key_code` is undefined in `vAmiga_keyboard.js` ?).

**Comment by mithrendal** [comment]
>Unmapped keys (e.g. F11) are interpreted as raw keycode 0

great catch 👍  ... fixed with #126  and pushed to gh-pages

**Comment by mithrendal** [comment]
@mras0 

>Light/dark mode is OK, but neither seems to match my preference. In light mode the blanked area is completely white, while the area outside the app seems to be more cream colored. In dark mode the blanked area is gray while I'd prefer it to be completely black. 

🤔 Hmmm maybe I should make this adjustable by the user ? I am not sure yet... what do you think ? 

but the area outside the app is definitely completely wrong, in your screen shot above and also here ...

see before 🙈:

![IMG_4197](https://user-images.githubusercontent.com/17108995/193423440-50c934dd-56d1-4a21-9538-9acbcf0a7b4d.jpeg)

in order to repair this for dark and light mode now the outside area is set to be completely black by https://vAmigaWeb.github.io ... hiding the notch ... I bet notch lovers will hate me now 😬 ...


see after 😎:

![image](https://user-images.githubusercontent.com/17108995/193423530-2c074141-f0cd-41ee-8e01-0901d301a6df.png)

also did this to https://vc64web.github.io

**Comment by mras0** [comment]
The area outside the screen isn't fully black for me in dark mode (same for vc64), but the background color extends fully to the edge now. I think that's fine, there's no need to complicate it more than dark/light mode. Again, I'd prefer it black but maybe there's some system setting overriding it?

**Comment by mithrendal** [comment]
>The area outside the screen...

@mras0 vAmigaWeb covers now that area... all space around notch or dynamic island is fully usable for example for the menu and the virtual touch joystick or even for the amiga screen buffer.

<img width="704" alt="image" src="https://github.com/vAmigaWeb/vAmigaWeb/assets/17108995/9080b616-9e11-4033-8e3a-25898879d34c">

**Comment by mras0** [comment]
Just tried it, and it seem to work very well! Watched Desert Dream to completion, and it behaved well. Quick question: How do I press "right mouse button" on my phone (to continue after disk1 in hardwired for example)?

**Comment by mithrendal** [comment]
>How do I press "right mouse button" on my phone

in the top right you find the two port selection boxes... the left is for port 1 and the right for port2

we need the one for port1

for pressing left mouse button choose either 

* the mouse pad (move|btn/btn) schema, which means left half of screen is mouse movements and right half is for the two buttons. Right mouse button = touch right top quadrant, left mouse button = touch right bottom quadrant

or

* touch joystick (move|fire) schema, which means left half for directions and right half of the screen for fire. Joystick fire closes the same electrical circuit as the left mouse button.

**Comment by mras0** [comment]
Ahh, thanks, works now! Thought I'd tried everything, but guess only displayed area counts for button presses? That was tripping me up a bit.

Maybe time to close this issue then? :)

**Comment by mithrendal** [comment]
>but guess only displayed area counts for button presses?

no definitely all of the screen area should be sensitiv to the press... or you are still on the old version ?

you could launch the Amitestkit (https://github.com/keirf/Amiga-Stuff/releases) and go to the controller section to watch the signals 

>Maybe time to close this issue then? :)

no I like still to review your suggestions on the top ... not finished yet 🤓

**Comment by mithrendal** [comment]
Oh no ... you were right ... touch joystick covers all the screen area but mouse touch pad is only restricted to the canvas ...

<img width="693" alt="image" src="https://github.com/vAmigaWeb/vAmigaWeb/assets/17108995/96589a4b-263c-48ae-8f1f-89a5f4d4ec56">

that is a bug ... will repair it

**Comment by mras0** [comment]
Was just about to reply, but yes using latest version (from 2023/09/12) and outside screen area clicks don't register. Good job otherwise!

**Comment by mithrendal** [comment]
fixed and pushed ... was only on canvas, now registers on complete screen

**Comment by mras0** [comment]
> fixed and pushed ... was only on canvas, now registers on complete screen #183
Yep, seems to work now :)

**Comment by mithrendal** [comment]
> In dark mode the blanked area is gray while I'd prefer it to be completely black.

BEFORE:

<img width="858" alt="image" src="https://github.com/vAmigaWeb/vAmigaWeb/assets/17108995/a3761a9c-d773-4d82-a533-3d8477019377">


AFTER:
<img width="858" alt="image" src="https://github.com/vAmigaWeb/vAmigaWeb/assets/17108995/ddb82513-19a4-4475-965b-353a85266863">


@mras0 this was what you meant right ?

**Comment by mras0** [comment]
> @mras0 this was what you meant right ? published to https://vamigaweb.github.io

Yes, exactly! Seems to work fine.
